components/org.wso2.carbon.identity.application.authenticator.iwa/src/test/java/org/wso2/carbon/identity/application/authenticator/iwa/IWAAuthenticationUtilTest.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.application.authenticator.iwa;
+
+import org.ietf.jgss.GSSException;
+import org.powermock.api.mockito.PowerMockito;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.carbon.user.core.claim.Claim;
+
+import java.io.File;
+import java.security.PrivilegedActionException;
+import javax.security.auth.login.LoginException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+
+public class IWAAuthenticationUtilTest {
+
+    String fullQualifiedUsername;
+    String password;
+    char[] passwordArray;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        fullQualifiedUsername = "testUser@carbon.super";
+        password = "password";
+        passwordArray = password.toCharArray();
+
+        System.setProperty("carbon.home", new File("src/test/resources/home").getAbsolutePath());
+    }
+
+    @Test (expectedExceptions = {IllegalArgumentException.class})
+    public void testGetDomainAwareUserName() {
+        String username = IWAAuthenticationUtil.getDomainAwareUserName(fullQualifiedUsername);
+        Assert.assertEquals(username, "testUser");
+
+        IWAAuthenticationUtil.getDomainAwareUserName("");
```

Comment:
Shall split the two cases into two separate test cases?

Reply:
Done